packages/synapse-react-client/src/components/styled/StyledTableContainer.tsx

```diff
@@ -23,6 +23,9 @@ export const StyledTableContainer: StyledComponent<StyledTableContainerProps> =
     ({ theme, density = 'default', noStripedRows = false }) => ({
       overflow: 'auto',
       paddingLeft: '2px',
+      ['thead > tr']: {
+        backgroundColor: theme.palette.grey[200],
+      },
```

Comment:
In addition to the `th` cells, add a background color to the entire header row. This prevents a visual regression where the full table width is wider than the sum of all columns (react-base-table auto-sized the columns to take up full width, so this wasn't an issue before).